main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/NativePropertyEditor/PropertyInfo/EnumDescriptorPropertyInfo.cs

```diff
@@ -70,34 +72,47 @@ internal override void SetValue<T> (object target, T value)
 		{
 			var tc = PropertyDescriptor.Converter;
 			if (tc.CanConvertFrom (typeof (T))) {
-				if (int.TryParse ((string)(object)value, out var index)) {
-					PropertyDescriptor.SetValue (PropertyProvider, values.GetValue (index));
-				} else {
-					throw new Exception (string.Format ("Error in SetValue<T> parsing {0} as integer in {1} (enum descriptor)", value, PropertyDescriptor.DisplayName));
+				try {
+					var item = tc.ConvertFrom (value);
+					PropertyDescriptor.SetValue (PropertyProvider, item);
+				} catch (Exception ex) {
+					LogInternalError ($"Error trying to set and convert a value: {value} T:{typeof (T).FullName}", ex);
+				}
+			} else {
+				base.SetValue<T> (target, value);
+			}
+		}
+
+		static int GetIndexOfValue (Array values, string value)
+		{
+			for (int i = 0; i < values.Length; i++) {
+				if (values.GetValue (i).ToString ().Equals (value)) {
+					return i;
 				}
 			}
+			return -1;
 		}
 
-		internal override Task<T> GetValueAsync<T> (object target)
+		internal async override Task<T> GetValueAsync<T> (object target)
 		{
 			//we need set the index of the selected item
 			T result = default;
 			try {
-				TypeConverter tc = PropertyDescriptor.Converter;
-				object cob = PropertyDescriptor.GetValue (PropertyProvider);
-				var index = Array.IndexOf (values, cob);
+				result = await base.GetValueAsync<T> (target);
```

Comment:
Unless we need to run everything on the UI thread:
```suggestion
				result = await base.GetValueAsync<T> (target).ConfigureAwait (false);
```

Reply:
Fixed